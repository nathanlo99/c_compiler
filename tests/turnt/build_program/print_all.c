
int wain(int *arr, int length) {
  int idx = 0;
  int sum = 0;
  while (idx < length) {
    println(*(arr + idx));
    sum = sum + *(arr + idx);
    idx = idx + 1;
  }
  return sum;
}
