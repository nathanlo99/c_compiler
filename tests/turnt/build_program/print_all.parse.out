procedures main
main INT WAIN LPAREN dcl COMMA dcl RPAREN LBRACE dcls statements RETURN expr SEMI RBRACE
INT int
WAIN wain
LPAREN (
dcl type ID
type INT STAR
INT int
STAR *
ID arr
COMMA ,
dcl type ID
type INT
INT int
ID length
RPAREN )
LBRACE {
dcls dcls dcl BECOMES NUM SEMI
dcls dcls dcl BECOMES NUM SEMI
dcls .EMPTY
dcl type ID
type INT
INT int
ID idx
BECOMES =
NUM 0
SEMI ;
dcl type ID
type INT
INT int
ID sum
BECOMES =
NUM 0
SEMI ;
statements statements statement
statements .EMPTY
statement WHILE LPAREN test RPAREN LBRACE statements RBRACE
WHILE while
LPAREN (
test sum LT sum
sum term
term factor
factor ID
ID idx
LT <
sum term
term factor
factor ID
ID length
RPAREN )
LBRACE {
statements statements statement
statements statements statement
statements statements statement
statements .EMPTY
statement PRINTLN LPAREN expr RPAREN SEMI
PRINTLN println
LPAREN (
expr test
test sum
sum term
term factor
factor STAR factor
STAR *
factor LPAREN expr RPAREN
LPAREN (
expr test
test sum
sum sum PLUS term
sum term
term factor
factor ID
ID arr
PLUS +
term factor
factor ID
ID idx
RPAREN )
RPAREN )
SEMI ;
statement expr SEMI
expr lvalue BECOMES expr
lvalue ID
ID sum
BECOMES =
expr test
test sum
sum sum PLUS term
sum term
term factor
factor ID
ID sum
PLUS +
term factor
factor STAR factor
STAR *
factor LPAREN expr RPAREN
LPAREN (
expr test
test sum
sum sum PLUS term
sum term
term factor
factor ID
ID arr
PLUS +
term factor
factor ID
ID idx
RPAREN )
SEMI ;
statement expr SEMI
expr lvalue BECOMES expr
lvalue ID
ID idx
BECOMES =
expr test
test sum
sum sum PLUS term
sum term
term factor
factor ID
ID idx
PLUS +
term factor
factor NUM
NUM 1
SEMI ;
RBRACE }
RETURN return
expr test
test sum
sum term
term factor
factor ID
ID sum
SEMI ;
RBRACE }
